MAUI/DataGrid/Grouping.md

```diff
@@ -0,0 +1,572 @@
+    //dataGrid.GroupColumnDescriptions.RemoveAt(0);
+}
+{% endhighlight%} 
+
+Run the application to render the following output: 
+
+![Remove groups in runtime in a DataGrid](Images\Grouping\Before_RemovingGroup.png)
+![Remove groups in runtime in a DataGrid](Images\Grouping\After_RemovingGrouping.png)
+
+N> Clear or remove grouping on [GridTapped event](), [GridDoubleTapped event](), or [GridLongPressed event]().
+
+## Events
+
+### GroupExpanding event
+
+The [SfDataGrid.GroupExpanding]() event occurs when group is being expanded.
+ 
+The [GroupChangingEventArgs]() of the `GroupExpanding` event provides the information about the expanding group and it has the following members:
+
+ `Syncfusion.Data.Group`: Gets expanded group.
+
+ `Cancel`: Decides to cancel group expansion.
+ 
+Cancel group expansion by setting [GroupChangingEventArgs.Cancel]() to `true`.
+
+{% tabs %}
+{% highlight c# %}
+this.dataGrid.GroupExpanding += dataGrid_GroupExpanding;
+
+void dataGrid_GroupExpanding(object sender, Syncfusion.SfDataGrid.XForms.GroupChangingEventArgs e)
+{
+    if (e.Group.Key.Equals(1001))    
+        e.Cancel = true;    
+}       
+{% endhighlight %}
+{% endtabs %}
+
+### GroupExpanded event
+
+The [SfDataGrid.GroupExpanded]() event occurs after group is expanded.
+
+The [GroupChangedEventArgs]() of the `GroupExpanded` event provides the information about the expanded group and it has the following member:
+
+ `Syncfusion.Data.Group`: Gets the expanded group.
+
+### GroupCollapsing event 
+
+The [SfDataGrid.GroupCollapsing]() event occurs when group is being collapsed.
+
+The [GroupChangingEventArgs]()  of the `GroupCollapsing` event provides the information about the collapsing group and it contains the following member:
+
+ `Syncfusion.Data.Group`: Gets collapsed group.
+
+ `Cancel`: Decides to cancel the group collapsing.
+
+Cancel the group is being collapsed by using the [GroupChangingEventArgs.Cancel]() of `GroupCollapsing` event.
+
+{% tabs %}
+{% highlight c# %}
+this.dataGrid.GroupCollapsing += dataGrid_GroupCollapsing;
+
+void dataGrid_GroupCollapsing(object sender, Syncfusion.SfDataGrid.XForms.GroupChangingEventArgs e)
+{
+    if (e.Group.Key.Equals(1001))    
+        e.Cancel = true;    
+}
+{% endhighlight %}
+{% endtabs %}
+
+### GroupCollapsed event
+ 
+The [SfDataGrid.GroupCollapsed]() event occurs after group collapsed.
+ 
+The [GroupChangedEventArgs]() of the `GroupCollapsed` event provides the information about collapsed group and it contains the following member:
+
+ `Syncfusion.Data.Group`: Gets the collapsed group.
+
+## How to hide the grouped column in SfDataGrid
+
+To hide/show a particular column gets grouped, set the [SfDataGrid.ShowColumnWhenGrouped]() property to `false`/`true`. Any column(s) added in the `GroupColumnDescriptions` collection will be shown or hidden based on the value of the [SfDataGrid.ShowColumnWhenGrouped]() property. Refer to the following code snippet:
```

Comment:
The visibility of the grouped column can be customized by the [SfDataGrid.ShowColumnWhenGrouped]() property. By default, all the grouped columns are visible.